packages/core/src/hydration/utils.ts

```diff
@@ -254,21 +271,65 @@ export function markRNodeAsClaimedByHydration(node: RNode, checkIfAlreadyClaimed
   if (!ngDevMode) {
     throw new Error(
         'Calling `markRNodeAsClaimedByHydration` in prod mode ' +
-        'is not supported and likely a mistake.');
+            'is not supported and likely a mistake.',
+    );
   }
   if (checkIfAlreadyClaimed && isRNodeClaimedForHydration(node)) {
     throw new Error('Trying to claim a node, which was claimed already.');
   }
-  (node as ClaimedNode).__claimed = true;
+  (node as HydratedNode).__hydration = {status: HydratedNodeStatus.Claimed};
   ngDevMode.hydratedNodes++;
 }
 
+export function markRNodeAsSkippedByHydration(node: RNode) {
+  if (!ngDevMode) {
+    throw new Error(
+        'Calling `markRNodeAsSkippedByHydration` in prod mode ' +
+            'is not supported and likely a mistake.',
+    );
+  }
+  (node as HydratedNode).__hydration = {status: HydratedNodeStatus.Skipped};
+  ngDevMode.componentsSkippedHydration++;
+}
+
+export function markRNodeAsHavingHydrationMismatch(
+    node: RNode,
+    expectedNode: string|null = null,
+    actualNode: string|null = null,
+) {
+  if (!ngDevMode) {
+    throw new Error(
+        'Calling `markRNodeAsMismatchedByHydration` in prod mode ' +
+            'is not supported and likely a mistake.',
+    );
+  }
+
+  // The RNode can be a standard HTMLElement
+  // The devtools component tree only displays Angular components & directives
+  // Therefore we attach the debug info to the closest a claimed node.
+  while (node && (node as HydratedNode).__hydration?.status !== 'claimed') {
+    node = node?.parentNode as RNode;
+  }
+
+
+  if (node) {
+    (node as HydratedNode).__hydration = {
+      status: HydratedNodeStatus.Mismatched,
+      expectedNodeDetails: expectedNode,
+      actualNodeDetails: actualNode,
```

Comment:
nit: we can rename function args (for consistency) and use shorter version:
```suggestion
      expectedNodeDetails,
      actualNodeDetails,
```